### PR TITLE
[SYCLomatic] Add comments and use compile time warning instead of runtime warning for querying momory clock and buswidth

### PR DESCRIPTION
**Pull request by wenhuiNi**
Signed-off-by: Ni, Wenhui <wenhui.ni@intel.com>

### CODE REVIEW

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 1]
```diff
@@ -31,6 +31,14 @@
 
 namespace dpct {
 
+inline void compiler_warning(const std::string &warning_msg) {
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]: please put this function in dpct::detail namespace.
rename it to:  warning

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 2]
```diff
@@ -117,7 +125,11 @@ class device_info {
   }
   size_t get_global_mem_size() const { return _global_mem_size; }
   size_t get_local_mem_size() const { return _local_mem_size; }
+  /// Returns the maximum clock rate of device's global memory in kHz. If
+  /// compiler not support this API then returns 3200000 kHz.
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
  /// compiler does not support this API then returns default value 3200000 kHz.
```

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 3]
```diff
@@ -117,7 +125,11 @@ class device_info {
...
   unsigned int get_memory_clock_rate() const { return _memory_clock_rate; }
+  /// Returns the maximum bus width between device and memory in bits. If
+  /// compiler not support this API then returns 64 bits.
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
  /// compiler does not support this API then returns default value 64 bits.
```

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 4]
```diff
@@ -185,7 +197,9 @@ class device_info {
   int _minor;
   int _integrated = 0;
   int _frequency;
-  unsigned int _memory_clock_rate = 0;
+  // Setting estimated value 3200000 kHz as memory_clock_rate default value.
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
  // Set estimated value 3200000 kHz as default value.
```

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 5]
```diff
@@ -185,7 +197,9 @@ class device_info {
...
+  unsigned int _memory_clock_rate = 3200000;
+  // Setting estimated value 64 bits as memory_bus_width default value.
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
  // Set estimated value 64 bits as default value.
```

**Comment by tangjj11** [comment]
the title has a wrong word `comemnts`.

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 6]
```diff
@@ -315,32 +329,21 @@ class device_ext : public sycl::device {
...
 #else
-    std::cerr
-        << "get_device_info: query memory_clock_rate and memory_bus_width are "
-           "not supported by the compiler you currently used."
-        << std::endl;
-    std::cerr << "memory_clock_rate default value is 0." << std::endl;
-    std::cerr << "memory_bus_width default value is 64." << std::endl;
+    compiler_warning(
+        "get_device_info: querying memory_clock_rate and memory_bus_width are "
+        "not supported by the compiler used. \nSetting 3200000 kHz as "
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp:344]:
```suggestion
        "not supported by the compiler used. \nSet 3200000 kHz as "
```

Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
        "not supported by the compiler used. \nUse 3200000 kHz as "
```

File: clang/test/dpct/helper_files_ref/include/device.hpp [hunk 7]
```diff
@@ -315,32 +329,21 @@ class device_ext : public sycl::device {
...
+        "memory_clock_rate default value.\nSetting 64 bits as memory_bus_width "
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/device.hpp]:
```suggestion
        "memory_clock_rate default value.\nUse 64 bits as memory_bus_width "
```

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by tangjj11 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**